lib/core/src/Cardano/Wallet/Api/Server.hs

```diff
@@ -4202,6 +4336,22 @@ instance IsServerError ErrReadRewardAccount where
                 , "wallets can do something with rewards and this one isn't."
                 ]
 
+instance IsServerError ErrReadPolicyPublicKey where
+    toServerError = \case
+        ErrReadPolicyPublicKeyNoSuchWallet e -> toServerError e
+        ErrReadPolicyPublicKeyNotAShelleyWallet ->
+            apiError err403 InvalidWalletType $ mconcat
+                [ "It is regrettable but you've just attempted an operation "
+                , "that is invalid for this type of wallet. Only new 'Shelley' "
+                , "wallets can do something with rewards and this one isn't."
+                ]
+        ErrReadPolicyPublicKeyAbsent ->
+            apiError err403 InvalidWalletType $ mconcat
+                [ "It seems the wallet lacks policy public key and minting/burning"
+                , "transaction cannot be realized. Please create wallet from mnemonics"
+                , "rather than account public key."
```

Comment:
```suggestion
            apiError err403 InvalidWalletType $ unwords
                [ "It seems the wallet lacks a policy public key. It's"
                , "therefore not possible to create a minting or burning"
                , "transaction. Please restore the wallet from a mnemonic"
                , "phrase instead of an account public key."
```